r/src/dataset.cpp

```diff
@@ -22,20 +22,26 @@
 // [[arrow::export]]
 ds::DataSourceDiscoveryPtr dataset___FSDSDiscovery__Make(
     const std::shared_ptr<fs::FileSystem>& fs,
-    const std::shared_ptr<fs::Selector>& selector) {
+    const std::shared_ptr<fs::Selector>& selector,
+    const std::shared_ptr<ds::PartitionScheme>& partition_scheme) {
   // TODO(npr): add format as an argument, don't hard-code Parquet
   auto format = std::make_shared<ds::ParquetFileFormat>();
+
   // TODO(fsaintjacques): Make options configurable
   auto options = ds::FileSystemDiscoveryOptions{};
+  if (partition_scheme != nullptr) {
```

Comment:
```suggestion
  if (!Rf_isNull(partition_scheme)) {
```

Reply:
No this was wrong, need two versions of the function, one with and one without partition_scheme arg. See r/src/array.cpp for example.

Reply:
Thanks, that's working